src/dune/cached_digest.ml

```diff
@@ -103,14 +103,23 @@ let refresh fn =
   let stats = Path.stat fn in
   refresh_ stats fn
 
+let chmod_error_reported = ref false
+
 let refresh_and_chmod fn =
   let stats = Path.stat fn in
   let () =
     (* We remove write permissions to uniformize behavior regardless of whether
        the cache is activated. No need to be zealous in case the file is not
        cached anyway. See issue #3311. *)
     if Cache.cachable stats.st_kind then
-      Path.chmod ~stats:(Some stats) ~mode:0o222 ~op:`Remove fn
+      try
+        Path.chmod ~stats:(Some stats) ~mode:0o222 ~op:`Remove fn
+      with _ ->
```

Comment:
This is one of those instances where you can catch `Sys.Break` when you didn't mean to. I think catching `Unix_error` is sufficient.

Reply:
Ah, didn't know about `Sys.break`. Fixed.